### DISCUSSION TITLE
Support `optimisticData` as a function

### DISCUSSION THREAD

**Discussion by ghost** [vercel/swr#1850] [Ideas]
First of all, thanks a lot for this library. I've been using it for almost a year, and more intensively for several months, and I love it!

Il like to create optimistic user interfaces, and the new `optimisticData` option with its auto-rollback behavior is great!

I'd like to push it a little further: could you support `optimisticData` as a function ? I think it would be consistent with the `mutate` first argument and with #1817.

This function would take the cached data as parameter, and would allow:
- using the global `mutate` function
- extracting and test a pure function that makes the mutation I want, depending on the current data and the operation that is processed

Again, thanks a lot for your work

**Comment by shuding** [comment]
Thanks for writing this, I created an issue to track the progress in #1858.

**Comment by shuding** [comment]
This feature is implemented in #1861 by @huozhi. 🎉

Will cut a beta release soon!

**Reply by ghost** [reply]
Thanks a lot! 🤩
